app/PermissionsContext.js

```diff
@@ -114,9 +114,9 @@ const PermissionsProvider = ({ children }) => {
 
   const requestLocationPermission = async () => {
     if (Platform.OS === 'ios') {
-      requestLocationForPlatform(PERMISSIONS.IOS.LOCATION_ALWAYS);
+      await requestLocationForPlatform(PERMISSIONS.IOS.LOCATION_ALWAYS);
```

Comment:
What is the reason for introducing this change? Was it to have the screen transition immediately? is this the intended behavior?

Reply:
Doing this makes the `requestLocationPermission` logic consistent with the `requestNotificationPermission` so that the app doesn't progress to the next screen until the user selects an option from the location permission popup that appears.